test/test_rlhf.py

```diff
@@ -164,6 +170,193 @@ def test_get_dataloader(
         assert not is_tensor_collection(dl)
 
 
+@pytest.mark.parametrize("batch_size", [5, 6])
+@pytest.mark.parametrize("block_size", [15, 50])
+@pytest.mark.parametrize("device", get_default_devices())
+def test_reward_model(tmpdir, batch_size, block_size, device):
+    dataset_name = f"{HERE}/datasets_mini/openai_summarize_tldr"
+    tensorclass_type = PromptData
+    dl = get_dataloader(
+        batch_size,
+        block_size,
+        tensorclass_type,
+        device,
+        dataset_name=dataset_name,
+        infinite=True,
+        prefetch=0,
+        split="train",
+        root_dir=tmpdir,
+        from_disk=True,
+    )
+
+    reward_model = GPT2RewardModel().to(device)
+
+    batch = next(dl)
+
+    rewards, end_scores = reward_model(
+        input_ids=batch.input_ids, attention_mask=batch.attention_mask
+    )
+
+    assert rewards.shape == torch.Size([batch_size, block_size])
+    assert end_scores.shape == torch.Size([batch_size])
+
+
+class TestRollout:
+    kl_coef = 0.1
+
+    @staticmethod
+    def init_transformer(
+        dropout=0.1,
+        device="cpu",
+        as_tensordictmodule=True,
+        inference=False,
+    ):
+        from transformers import GPT2LMHeadModel
+
+        model = GPT2LMHeadModel(GPT2Config())
+        model.to(device)
+
+        if as_tensordictmodule:
+            model = TensorDictModule(
+                model,
+                in_keys={
+                    "input_ids": "input_ids",
+                    "attention_mask": "attention_mask",
+                    "labels": "labels",
+                },
+                out_keys=["logits"] if inference else ["loss", "logits"],
+            )
+        return model
+
+    def init_reward_model(self, device=None):
+        model = GPT2RewardModel()
+        model.to(device)
+
+        model = TensorDictModule(
+            model,
+            in_keys=["input_ids", "attention_mask"],
+            out_keys=["rewards", "end_scores"],
+        )
+        return model
+
+    @property
+    def _dummy_batch(self):
+        return PromptData.from_tensordict(
+            TensorDict.load_memmap(f"{HERE}/datasets_mini/tldr_batch")
+        )
```

Comment:
why not staticmethod?
```suggestion
    @staticmethod
    def _get_dummy_batch():
        return PromptData.from_tensordict(
            TensorDict.load_memmap(f"{HERE}/datasets_mini/tldr_batch")
        )
```